kotlin-inject-compiler-core/src/main/kotlin/me/tatarka/inject/compiler/InjectGenerator.kt

```diff
@@ -214,6 +217,9 @@ fun AstMethod.isProvider(): Boolean =
         is AstProperty -> true
     } && receiverParameterType == null && returnType.isNotUnit()
 
+fun AstClass.toInjectName(): String =
+    "Inject${asClassName().simpleNames.joinToString("_")}"
```

Comment:
How does this work with class names with backticks?

Reply:
Backticks work in variable names too so I think it still works?

Reply:
Or maybe not because you'd get 'Inject\`My backtick class`'

Reply:
https://github.com/evant/kotlin-inject/issues/113